### ISSUE TITLE
Public shelves missing again

### ISSUE THREAD

**Issue by wgibcus** [janeczku/calibre-web#2017]
**Describe the bug/problem**
After updating to 0.6.12 from 0.6.9 the public shelves are no longer available to logged on users or to guest access. The shelves are blank despite still seeing the shelves in the "Book Details" windows for each book. The issue appears to be a display issue.

It was all working beforehand so it's definitely not a setup issue.

--

<br class="Apple-interchange-newline">

Program Library	Installed Version
Platform	Linux 5.4.0-1022-raspi #25-Ubuntu SMP PREEMPT Thu Oct 15 13:31:49 UTC 2020 aarch64 aarch64
Python	3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
Calibre_Web	0.6.12 - 54d06e580d73ba58274e7ed4efbd8fc94b7e1856 - 2021-05-22T14:08:46+02:00
WebServer	Gevent 20.9.0
Flask	1.1.2
Flask_Login	0.5.0
Flask_Principal	0.4.0
Werkzeug	1.0.1
Babel	2.8.0
Jinja2	2.10.1
Requests	2.22.0
SqlAlchemy	1.3.17
pySqlite	2.6.0
SQLite	3.31.1
iso639	0.4.5
pytz	2019.3
Unidecode	installed
Goodreads	installed
jsonschema	3.2.0
flask_dance	3.0.0
greenlet	0.4.17
Image Magick	ImageMagick 6.9.10-23 Q16 aarch64 20190101 https://imagemagick.org
PyPdf	v1.26.0
lxml	v4.5.0.0
Wand	0.5.9
Comic_API	installed
ebook converter	not installed
unrar	UNRAR 5.61 beta 1 freeware Copyright (c) 1993-2018 Alexander Roshal
kepubify	not installed

**Comment by OzzieIsaacs** [comment]
For me it' working:
![grafik](https://user-images.githubusercontent.com/6438895/121390065-4c9db200-c94d-11eb-92c8-dbc7f7c7e8d1.png)


Please send me the debug package (available in admin section)

**Comment by wgibcus** [comment]
> For me it' working:
> ![grafik](https://user-images.githubusercontent.com/6438895/121390065-4c9db200-c94d-11eb-92c8-dbc7f7c7e8d1.png)
> 
> Please send me the debug package (available in admin section)

I have another instance which is working but on my main site, it is not. The upgrade definitely caused the issue as all was well before this.

**Comment by wgibcus** [comment]
Further to this, all of this code is missing from the broken shelves:
`

&lt;a id="order_shelf" href="/shelf/order/2" class="btn btn-primary"&gt;Arrange books manually &lt;/a&gt;
--
&nbsp; | &lt;button id="toggle_order_shelf" type="button" data-alt-text="Disable Change order" class="btn btn-primary"&gt;Enable Change order&lt;/button&gt;
&nbsp; | &lt;div class="filterheader hidden-xs hidden-sm"&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort according to book date, newest first" id="new" class="btn btn-primary disabled" href="/shelf/2/new"&gt;&lt;span class="glyphicon glyphicon-book"&gt;&lt;/span&gt; &lt;span class="glyphicon glyphicon-calendar"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-order"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort according to book date, oldest first" id="old" class="btn btn-primary disabled" href="/shelf/2/old"&gt;&lt;span class="glyphicon glyphicon-book"&gt;&lt;/span&gt; &lt;span class="glyphicon glyphicon-calendar"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-order-alt"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort title in alphabetical order" id="asc" class="btn btn-primary disabled" href="/shelf/2/abc"&gt;&lt;span class="glyphicon glyphicon-font"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-alphabet"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort title in reverse alphabetical order" id="desc" class="btn btn-primary disabled" href="/shelf/2/zyx"&gt;&lt;span class="glyphicon glyphicon-font"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-alphabet-alt"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort authors in alphabetical order" id="auth_az" class="btn btn-primary disabled" href="/shelf/2/authaz"&gt;&lt;span class="glyphicon glyphicon-user"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-alphabet"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort authors in reverse alphabetical order" id="auth_za" class="btn btn-primary disabled" href="/shelf/2/authza"&gt;&lt;span class="glyphicon glyphicon-user"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-alphabet-alt"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort according to publishing date, newest first" id="pub_new" class="btn btn-primary disabled" href="/shelf/2/pubnew"&gt;&lt;span class="glyphicon glyphicon-calendar"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-order"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;a data-toggle="tooltip" title="Sort according to publishing date, oldest first" id="pub_old" class="btn btn-primary disabled" href="/shelf/2/pubold"&gt;&lt;span class="glyphicon glyphicon-calendar"&gt;&lt;/span&gt;&lt;span class="glyphicon glyphicon-sort-by-order-alt"&gt;&lt;/span&gt;&lt;/a&gt;
&nbsp; | &lt;/div&gt;

`

**Comment by OzzieIsaacs** [comment]
If this is not displayed then the no. of books in the shelf is zero

**Comment by OzzieIsaacs** [comment]
I had a look at the logbooks. There are hundres of messages "Cannot find book entry xxx", which leads to the conclusion that the calibre database has some problem (missing entries for book file data).  
There are also errors with message: `sqlite3.IntegrityError: UNIQUE constraint failed: authors.name`. This leads to the conclusion  that the calibre database has some problem (dublicate entries for authors).  
Could you please run a check of the metadata.db with calibre (Library -> check libary)

**Comment by wgibcus** [comment]
> I had a look at the logbooks. There are hundres of messages "Cannot find book entry xxx", which leads to the conclusion that the calibre database has some problem (missing entries for book file data).
> There are also errors with message: `sqlite3.IntegrityError: UNIQUE constraint failed: authors.name`. This leads to the conclusion that the calibre database has some problem (dublicate entries for authors).
> Could you please run a check of the metadata.db with calibre (Library -> check libary)

Yes, did that. I fixed all the errors using Calibre but the same error is apparent. I did a fresh install and connected the working metadata.db and I could add a shelf and books to that shelf. I connected the "fixed" metadata.db from the other site and the same problem appeared.

The error is definitely in the metadata.db. I'm looking into the DB now so any assistance on how the metadata.db talks to the Calibre-web app.db would be greatly appreciated. :)

**Comment by wgibcus** [comment]
Ok I've solved this now. There was an erroneous table in the metadata.db file called "books_custom_column_1", which was a hangover from the days when I used Calibre. Deleting this table made contents of the shelves reappear. 

To work this out, I simply compared the working metadata.db to the broken one and cleaned up all the tables that didn't appear in the working database. Some of these were Calibre settings and I am assuming that they will be written back by Calibre if I ever use it again in the future.
